luigi/scheduler.py

```diff
@@ -234,6 +234,27 @@ def load(self):
         else:
             logger.info("No prior state file exists at %s. Starting with clean slate", self._state_path)
 
+    def cascade_disabled_state(self, tasks):
+
+        """ Iterate through all pending tasks, setting the status to DISABLED
+        if at least one of it's dependencies is DISABLED and the rest are
+        either DONE or DISABLED. """
+
+        for task in tasks:
```

Comment:
I don't see how this will always work? If `tasks` isn't topologically sorted then you might not disable all tasks